tools/api_proto_breaking_change_detector/detector.py

```diff
@@ -0,0 +1,147 @@
+    def lock_file_changed(self):
+        pass
+
+
+class ChangeDetectorError(Exception):
+    pass
+
+
+class ChangeDetectorInitializeError(ChangeDetectorError):
+    pass
+
+
+BUF_LOCK_FILE = "tmp.json"
+
+
+class BufWrapper(ProtoBreakingChangeDetector):
+
+    def __init__(self, path_to_before, path_to_after, additional_args=None):
+        if not Path(path_to_before).is_file():
+            raise ValueError(f"path_to_before {path_to_before} does not exist")
+
+        if not Path(path_to_after).is_file():
+            raise ValueError(f"path_to_after {path_to_after} does not exist")
+
+        self.path_to_before = path_to_before
+        self.path_to_after = path_to_after
+        self.additional_args = additional_args
+
+    def run_detector(self):
+        # 1) pull buf BSR dependencies with buf mod update (? still need to figure this out. commented out for now)
+        # 2) copy buf.yaml into temp dir
+        # 3) copy start file into temp dir
+        # 4) buf build -o tmp_file
+        # 5) copy changed file into temp dir
+        # 6) buf breaking --against tmp_file
+        # 7) check for differences (if changes are breaking, there should be none)
+
+        temp_dir = tempfile.TemporaryDirectory(
+            prefix=str(Path(".").absolute())
+            + os.sep)  # buf requires protobuf files to be in a subdirectory of cwd
+        buf_path = runfiles.Create().Rlocation("com_github_bufbuild_buf/bin/buf")
+
+        buf_config_loc = Path(".", "tools", "api_proto_breaking_change_detector")
+
+        #copyfile(Path(buf_config_loc, "buf.lock"), Path(".", "buf.lock")) # not needed? refer to comment below
+        yaml_file_loc = Path(".", "buf.yaml")
+        copyfile(Path(buf_config_loc, "buf.yaml"), yaml_file_loc)
+
+        # TODO: figure out how to automatically pull buf deps
+        # `buf mod update` doesn't seem to do anything, and the first test will fail because it forces buf to automatically start downloading the deps
+        #        bcode, bout, berr = run_command(f"{buf_path} mod update")
+        #        bout, berr = ''.join(bout), ''.join(berr)
+
+        target = Path(temp_dir.name, f"{Path(self.path_to_before).stem}.proto")
+
+        buf_args = [
+            "--path",
+            str(target.relative_to(
+                Path(".").absolute())),  # buf requires relative pathing for roots...
+            "--config",
+            str(yaml_file_loc),
+        ]
+        if self.additional_args is not None:
+            buf_args.extend(self.additional_args)
+
+        copyfile(self.path_to_before, target)
+
+        initial_code, initial_out, initial_err = run_command(
+            ' '.join([buf_path, f"build -o {Path(temp_dir.name, BUF_LOCK_FILE)}", *buf_args]))
+        initial_out, initial_err = ''.join(initial_out), ''.join(initial_err)
+
+        if initial_code != 0 or len(initial_out) > 0 or len(initial_err) > 0:
+            raise ChangeDetectorInitializeError("Unexpected error during init")
```

Comment:
Please add more info to the exception (from a user's perspective, seeing this error doesn't give them anything to change, other than debugging this code)